docs/api/apiv3/components/schemas/attachment_model.yml

```
@@ -27,11 +24,19 @@ properties:
     allOf:
     - $ref: "./formattable.yml"
     - description: A user provided description of the file
+  status:
+    type: string
+    enum:
+    - uploaded
+    - prepared
+    - scanned
+    - quarantined
+    - rescan
   contentType:
     type: string
     description: The files MIME-Type as determined by the server
   digest:
-    type: string
+    type: [string, null]
```

🟡 for OpenAPIv3.1 the better syntax for an array of types is:

```yaml
type:
  - string
  - "null"
```

e.g. read [here](https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0) or check the examples [here](https://spec.openapis.org/oas/v3.1.0.html). in YAML the arrays mostly are written without `[]`.

---

I have to admit that I find the single-line syntax easier to read for "something or null". This is a YAML feature, not an OpenAPI feature.

However, I see a valid remark in my use of `null`, which is interpreted as a literal null value when I intended to put the string `"null"`. So I'd probably suggest going for full JSON-mode:

```suggestion
    type: ["string", "null"]
```

WDYT?

---

yes, the problem with the literal is something you MUST fix. still, the flow syntax for arrays is not the default syntax.

Usually it is preferred to use the block syntax for arrays, e.g. because of
- easier readable for longer lists or list with objects
- easier to diff/merge with git
- it keeps the indented YAML style and thus provides an easier flow when reading YAMLs

the flow syntax is usually only used when for some reasons it is important to keep the YAML compact.

As the API spec is something that is usually parsed by machines, I prefer if I can see syntax errors in a quick glance over a file without reading every line in detail, and that is way easier with the block syntax.

---

will change

---

changed